### PR TITLE
adds code for safe return in case of no impostors for lmnn

### LINKED ISSUE #17: Error when impostors list is empty
Numpy throws an error in np.hstack(impostors) when impostors is empty.
What would be the recommended way to fix this?

### PR DESCRIPTION
**Pull request by anirudt**
Attempt to fix #17 .

I have done it for LMNN, is there a need for the others?

### CODE REVIEW

File: metric_learn/lmnn.py [hunk 1]
```diff
@@ -203,6 +206,9 @@ def _find_impostors(self, furthest_neighbors):
         tmp = np.ravel_multi_index((i,j), shape)
         i,j = np.unravel_index(np.unique(tmp), shape)
       impostors.append(np.vstack((in_inds[j], out_inds[i])))
+    if len(impostors) == 0:
+        """No impostors detected"""
+        return []
```
Comment (perimosocordiae) [inline metric_learn/lmnn.py]: No need for a docstring here. Also, you can just return `impostors` instead of returning a new empty list.

**Comment by perimosocordiae** [comment]
Thanks! One small comment, but the change looks good overall. If something similar needs to be done for the other methods, we can handle those fixes separately.

File: metric_learn/lmnn.py (hunk 1 above)
Comment (anirudt) [inline metric_learn/lmnn.py]: Sure, will do.

**Comment by anirudt** [comment]
Sure, @perimosocordiae . Looking at the CI Build Details, the RCA test hasn't gone through. Does it need  a look into it?

**Comment by perimosocordiae** [comment]
The test failure is unrelated, though it did make me smile: `0.25054383490356519 not less than 0.25`. So close!

Thanks for your contribution!

**Pull request merged**